src/snowflake/snowpark/session.py

```diff
@@ -645,9 +646,10 @@ def auto_clean_up_temp_table_enabled(self) -> bool:
         The default value is ``False``.
```

Comment:
The doctoring says:
> When setting this parameter to ``True``, Snowpark will automatically clean up temporary tables created by
        :meth:`DataFrame.cache_result` in the current session when the DataFrame is no longer referenced (i.e., gets garbage collected)
        
Do we make any distinction between calls to `cache_result` made by the user and those made by the system (e.g., during large query breakdown or repeated subquery elimination, or possibly as part of future optimizations)?

Reply:
Discussed offline. Resolving.